### PR TITLE
Enable wasm light clients

### PR DESCRIPTION
**Pull request by ash-burnt**

### CODE REVIEW

**Review by 2xburnt — APPROVED** [review]

File: app/app.go [hunk 1]
```diff
@@ -704,6 +713,20 @@ func NewWasmApp(
 	wasmOpts = append(owasm.RegisterStargateQueries(*app.GRPCQueryRouter(), appCodec), wasmOpts...)
 	wasmOpts = append(wasmOpts, tokenFactoryOpts...)
 
+	// instantiate the Wasm VM with the chosen parameters
+	wasmVM, err := wasmvm.NewVM(
+		wasmDir,
+		availableCapabilities,
+		WasmContractMemoryLimit, // default of 32
```
Comment (edjroz) [inline app/app.go:720]: I'm gonna sound like a teacher, but 32 Bytes? KB? MB? GB?

**Review by edjroz — DISMISSED** [review]

**Review by edjroz — CHANGES_REQUESTED** [review]
Integration test needs to mark version we are going to in the file `integration_tests/upgrade_ibc_test.go`

```
...
xionVersionTo   = "<correct_sha>"
xionUpgradeName = "v12"
```
as it is a constant to keep up with the tests accross upgrades.

**Comment by ash-burnt** [comment]
@edjroz I updated the upgrade version, but there isn't a correct sha to set it to, because the heighliner CI isn't building releases currently. Once it is live again, it will be doable

**Review by edjroz — APPROVED** [review]

**Review by justinbarry — APPROVED** [review]
LGTM

**Pull request merged**